tests/zquantum/core/estimation/estimation_test.py

```diff
@@ -685,11 +760,87 @@ def test_calculate_exact_expectation_values(
             expectation_values_list, target_expectations, estimation_tasks
         ):
             assert len(expectation_values.values) == len(task.operator.terms)
-            # TODO: add tests for correlations and covariances
             np.testing.assert_array_almost_equal(
                 expectation_values.values, target.values
             )
 
+    @pytest.mark.parametrize(
+        "estimation_tasks,target_expectations", TEST_CASES_EIGENSTATES
+    )
+    def test_covariance_and_correlations_when_averaging_for_eigenstates(
+        self, simulator, estimation_tasks, target_expectations
+    ):
+        expectation_values_list = estimate_expectation_values_by_averaging(
+            simulator, estimation_tasks
+        )
+        for expectation_values, target, task in zip(
+            expectation_values_list, target_expectations, estimation_tasks
+        ):
+            for cov, corr, target_corr, target_cov in zip(
+                expectation_values.estimator_covariances,
+                expectation_values.correlations,
+                target.correlations,
+                target.estimator_covariances,
+            ):
+                assert len(corr) == len(task.operator.terms)
+                assert len(corr) == len(task.operator.terms)
```

Comment:
These two lines are duplicated. Did you mean to assert on `cov`?
```suggestion
                assert len(corr) == len(task.operator.terms)
                assert len(cov) == len(task.operator.terms)
```

Reply:
Yup!